pkg/virt-launcher/virtwrap/access-credentials/access_credentials.go

```diff
@@ -320,12 +315,22 @@ func (l *AccessCredentialManager) agentSetAuthorizedKeys(domName string, user st
 		// Zero flags argument means that the authorized_keys file is overwritten with the authorizedKeys
 		return domain.AuthorizedSSHKeysSet(user, authorizedKeys, 0)
 	}()
-	if errors.Is(err, libvirt.ERR_NO_SUPPORT) {
-		// If AuthorizedSSHKeysSet method is not supported, use the old method
-		desiredAuthorizedKeys := strings.Join(authorizedKeys, "\n")
-		return l.agentWriteAuthorizedKeysFile(domName, user, desiredAuthorizedKeys)
+	if firstErr == nil {
+		return nil
 	}
-	return err
+
+	// If AuthorizedSSHKeysSet method failed, use the old method
+	desiredAuthorizedKeys := strings.Join(authorizedKeys, "\n")
+	secondErr := l.agentWriteAuthorizedKeysFile(domName, user, desiredAuthorizedKeys)
+	if secondErr == nil {
+		return nil
+	}
+
+	return fmt.Errorf(
+		"failed to set SSH keys: error from libvirt's AuthorizedSSHKeysSet(): %w; error from using guest-file-write: %w",
+		firstErr,
+		secondErr,
+	)
```

Comment:
supernit - really not important but you could throw this in a const and assert in the unit tests..

```suggestion
const setAuthorizedKeysErrMsgFmt = "failed to set SSH keys: error from libvirt's AuthorizedSSHKeysSet(): %w; error from using guest-file-write: %w"
    [..]
func (l *AccessCredentialManager) agentSetAuthorizedKeys(domName string, user string, authorizedKeys []string) error {
    [..]
	return fmt.Errorf(setAuthorizedKeysErrMsgFmt, firstErr, secondErr)
```

Reply:
I prefer it as it is now. The unit tests checks only a substring of the error message.